docs/haskell-use-cases.rst

```diff
@@ -360,3 +360,126 @@ Then, the user will add ``--define use_worker=True`` in the command line when ca
+
+  nixpkgs_package(
+      name = "raw-haskell-base-image",
+      repository = "//nixpkgs:default.nix",
+      nix_file = "//nixpkgs:haskellBaseImageDocker.nix", # See below for how to define this
+      build_file_content = """
+  package(default_visibility = [ "//visibility:public" ]),
+  exports_file(["image"])
+      """,
+  )
+
+And finally use the ``rules_docker`` ``container_load`` functionality to grab the docker image built by the previous ``raw-haskell-base-image`` target ::
+
+  container_load(
+      name = "haskell-base-image",
+      file = "@raw-haskell-base-image//:image",
+  )
+
+Step two requires that we specify our nixpkgs/haskellBaseImageDocker.nix file as follows ::
+
+  # nixpkgs is provisioned by rules_nixpkgs for us which we set to be ./default.nix
+  with import <nixpkgs> { system = "x86_64-linux"; };
+
+  # Build the base image.
+  # The output of this derivation will be a docker archive in the same format as
+  # the output of `docker save` that we can feed to
+  # [container_load](https://github.com/bazelbuild/rules_docker#container_load)
+  let
+    haskellBase = dockerTools.buildLayeredImage {
+      name = "haskell-base-image-unwrapped";
+      created = "now";
+      contents = [ glibc libffi gmp zlib iana-etc cacert ]; # Here we can specify nix-provisioned libraries our haskell_binary products may need at runtime
+    };
+    # rules_nixpkgs require the nix output to be a directory,
+    # so we create one in which we put the image we've just created
+  in runCommand "haskell-base-image" { } ''
+    mkdir -p $out
+    gunzip -c ${haskellBase} > $out/image
+  ''
+
+Step three pulls all this together in a build file to actually assemble our final docker image. In a BUILD.bazel file, we'll need the following ::
+
+  load("@io_bazel_rules_docker//cc:image.bzl", "cc_image")
+  load("@io_bazel_rules_docker//container:container.bzl", "container_push")
+
+  haskell_binary(
+      name = "my_binary,
+      srcs = ["Main.hs"],
+      compiler_flags = [
+          "-O2",
+          "-threaded",
+          "-rtsopts",
+          "-with-rtsopts=-N",
+      ],
+      deps = [
+          ":my_haskell_library_dep", # for example...
+          # ...
+      ],
+  )
+
+  cc_image(
+      name = "my_binary_image",
+      base = "@haskell-base-image//image",
+      binary = ":my_binary",
+      ports = [ "8000/tcp" ],
+      creation_time = "{BUILD_TIMESTAMP}",
+      stamp = True,
+  )
+
+And you may want to use ``rules_docker`` to push your docker image as follows ::
+
+  container_push(
+      name = "my_binary_push",
+      image = ":my_binary_image",
+      format = "Docker",
+      registry = "gcr.io", # For example using a GCP GCR repository
+      repository = "$project-name-here/$my_binary_image_label",
+      tag = "{BUILD_USER}",
+ )
+
+*n.b* This obviously requires a linux build machine!
```

Comment:
Why doesn't this work on macOS?

Reply:
You can't build haskell base images in nix on darwin. This is a known issue with darwin for nix dockerTools. 

In the nix file snippet in the code I have `system = "x86_64-linux";`

Reply:
> This is a known issue with darwin for nix dockerTools.

Is there an issue on the Nix issue tracker? I think it would be good to link to it from the docs and rephrase to say that it requires a Linux builder because of that issue.

Reply:
I'll add this link and repush in an hour or so https://github.com/NixOS/nixpkgs/issues/16696

Reply:
I've pushed the rewording with the hyperlink @aherrmann